jetty-ee10/jetty-ee10-annotations/src/main/java/org/eclipse/jetty/ee10/annotations/AnnotationParser.java

```diff
@@ -917,7 +917,7 @@ public void resetParsedClasses()
      *
      * @param name the class file name
      * @return whether the class file name is valid
-     * TODO: does multiple things, we should be able to move this FileID in future PR
+     * TODO: does multiple things, we should be able to eliminate this in future PR by consolidating the ZipFS and Directory scanning behaviors into a single Files.walk() stream
```

Comment:
Do you already have the shape of that future PR in your head?

Reply:
Yes.

I want to consolidate the WAR/JAR file scanning/walking with the Directory scanning/walking.
AFAICT, the only difference between the two is that the Directory walking technique also does a "is hidden file" check, which we have a static predicate for in `FileID`, and that same check in WAR walk is harmless.   That means the AnnotationParser gets even simpler.